server/src/ai/ai.router.ts

```diff
@@ -1,39 +1,98 @@
 import express from 'express';
 import { CourseIdRequestType } from '../course/course.type';
-import { makeSummary, getCoursesWithMinReviews, getReviewsForSummary } from './ai.functions';
-
+import { getCoursesWithMinReviews, getReviewsForSummary, summarize, updateCourseWithAI } from './ai.functions';
 const aiRouter = express.Router();
-
 aiRouter.use(express.json());
 
-/** Reachable at POST /api/ai/summarizeReviews
+/** Reachable at POST /api/ai/update-all-courses
+ * retrieves all course IDs with a minimum of 3 reviews,
+ * and then attempts to update the classSummary, classTags, and freshness
+ * for each course ID by calling updateCoursesWithAI.
+ * returns a message summarizing how many courses were updated successfully
+ * and how many failed.
+ */
+aiRouter.post('/update-all-courses', async (req, res) => {
```

Comment:
I think the route name could be a bit clearer, something like `/api/ai/summarize-courses/` maybe?